drivers/adc-dac/ad5592r/ad5592r.c

```diff
@@ -122,12 +128,12 @@ int32_t ad5592r_read_adc(struct ad5592r_dev *dev, uint8_t chan,
 	if (ret < 0)
 		return ret;
 
-	ret = ad5592r_spi_wnop_r16(dev, &dev->spi_msg);
-	if (ret < 0)
-		return ret;
-
-	*value = dev->spi_msg;
-
+	for (i = 0; i < num_chans; i++) {
+		ret = ad5592r_spi_wnop_r16(dev, &dev->spi_msg);
```

Comment:
one single spi_write_and_read call should be used for multiple channel read. No need for the for loop.

Reply:
I'm not sure how that would work, would it not be necessary to call multiple SPI frame reads to capture the data being generated? For instance, if 5 channels were to be read back, 5 SPI reads would need to be occur for that data to be captured and then stored in 5 elements of the array. One SPI read call would only read back a single channel. See datasheet excerpt for 2 channel readback on the AD5592R (page 33).

![AD5592R_Multichannel_noRepeat](https://user-images.githubusercontent.com/56347769/82916907-94cc7b00-9f6a-11ea-9278-ed3511bb2fd0.png)

Reply:
This is the function:
```
int32_t spi_write_and_read(struct spi_desc *desc,
			   uint8_t *data,
			   uint16_t bytes_number)
```
You can send the number of bytes you want to fulfill your purpose for the spi transfer. In your case 5 channels (16 bits each) -> bytes_number = 5 * 2 = 10 bytes. Just make sure our data array is large enough to store all that data.

Reply:
I don't think that will work after all. The conversations for the AD5593/2R only occur on the falling edge of the CS pin. Using a single spi_write_and_read will only cause the CS pin to fall once - meaning that even given the right size array to store the data, only data from one channel will be collected. 

![spi_read_and_write_ADC_read](https://user-images.githubusercontent.com/56347769/83497989-33993000-a4b3-11ea-8e07-5ff92780845a.png)

![AD5592R_CS](https://user-images.githubusercontent.com/56347769/83498276-97bbf400-a4b3-11ea-8578-6187adb21711.png)

This is a reupload - my original comment seems to have disappeared when I updated the pull request.

Reply:
You are correct. I missed this when checking the datasheet, it needs a SYNC to do the conversion for the next channel. Then you can do it with multiple spi_read_writes.

Reply:
I have separated single and multiple channel reads into two functions, is there a need to make any other changes?

Reply:
I think this is ok. Will check the rest of the modifications.